core/src/main/java/hudson/model/Actionable.java

```diff
@@ -72,16 +76,26 @@
      *             May still be called for compatibility reasons from subclasses predating {@link TransientActionFactory}.
      */
     @Deprecated
-	public List<Action> getActions() {
-		if(actions == null) {
-			synchronized (this) {
-				if(actions == null) {
-					actions = new CopyOnWriteArrayList<Action>();
-				}
-			}
-		}
-		return actions;
-	}
+    public List<Action> getActions() {
+        return getOrCreateActions();
+    }
+
+    /**
+     * We need to handle the initialization of the actions list in Actionable so that child classes that override
+     * getActions() for historical reasons do not have to override the manipulation methods: {@link #addAction(Action)},
+     * {@link #replaceAction(Action)}, {@link #removeAction(Action)}, etc.
```

Comment:
I think this was an incompatible change: [JENKINS-39555](https://issues.jenkins-ci.org/browse/JENKINS-39555). `FlowNode.getActions()` was deliberately overridden to control storage.